src/data/set/basic.lean

```diff
@@ -132,7 +137,34 @@ assume h : x ∈ ∅, h
 @[simp] theorem not_not_mem [decidable (a ∈ s)] : ¬ (a ∉ s) ↔ a ∈ s :=
 not_not
 
-/- empty set -/
+/-! ### Non-empty sets -/
+
+/-- A set `s` is not empty. -/
```

Comment:
An extended docstring explaining the purpose of this would be welcome here. Something like
```lean
/-- The property `s.nonempty` expresses the fact that the set `s` is not empty. It should be used in theorem assumptions instead of `∃ x, x ∈ s` or `s ≠ ∅` as it gives access to a nice API thanks to the dot notation. -/ 
```

Reply:
Copied from your comment. Thank you!